src/services/variables.ts

```diff
@@ -16,26 +25,49 @@ export const VAR_SERVICE = 'service';
 export const VAR_SERVICE_EXPR = '${service}';
 export const VAR_DATASOURCE = 'ds';
 export const VAR_DATASOURCE_EXPR = '${ds}';
+export const MIXED_FORMAT_EXPR = `| json | logfmt | drop __error__, __error_details__`;
+export const JSON_FORMAT_EXPR = `| json | drop __error__, __error_details__`;
+export const LOGS_FORMAT_EXPR = `| logfmt`;
+// This variable is hardcoded to the value of MIXED_FORMAT_EXPR. This is a hack to get logs context working, we don't want to use a variable for a value that doesn't change and cannot be updated by the user.
 export const VAR_LOGS_FORMAT = 'logsFormat';
 export const VAR_LOGS_FORMAT_EXPR = '${logsFormat}';
 export const VAR_LINE_FILTER = 'lineFilter';
 export const VAR_LINE_FILTER_EXPR = '${lineFilter}';
-export const LOG_STREAM_SELECTOR_EXPR = `{${VAR_LABELS_EXPR}} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR} ${VAR_LEVELS_EXPR} ${VAR_FIELDS_EXPR} ${VAR_LINE_FILTER_EXPR}`;
+export const LOG_STREAM_SELECTOR_EXPR = `{${VAR_LABELS_EXPR}} ${VAR_PATTERNS_EXPR} ${VAR_LINE_FILTER_EXPR} ${VAR_LEVELS_EXPR} ${VAR_LOGS_FORMAT_EXPR} ${VAR_FIELDS_EXPR}`;
 export const PATTERNS_SAMPLE_SELECTOR_EXPR = `{${VAR_LABELS_EXPR}} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR}`;
 export const EXPLORATION_DS = { uid: VAR_DATASOURCE_EXPR };
 export const ALL_VARIABLE_VALUE = '$__all';
 export const LEVEL_VARIABLE_VALUE = 'detected_level';
-export const PATTERNS_TEXT_FILTER = 'patternsFilter';
 export const SERVICE_NAME = 'service_name';
 export const EMPTY_VARIABLE_VALUE = '""';
 
-export function getLogsStreamSelector(options: {
+export type ParserType = 'logfmt' | 'json' | 'mixed' | '';
```

Comment:
I'm not sure I understand the empty parser. Should maybe be `none`?

Reply:
swapped out for "structuredMetadata" should be a bit more clear anyway